paddle/fluid/operators/batch_norm_op.cc

```diff
@@ -169,6 +170,32 @@ framework::OpKernelType BatchNormOp::GetExpectedKernelType(
                                  library);
 }
 
+framework::OpKernelType BatchNormOp::GetKernelTypeForVar(
+    const std::string &var_name, const Tensor &tensor,
+    const framework::OpKernelType &expected_kernel_type) const {
+#ifdef PADDLE_WITH_MKLDNN
+  // Only input require reshaping, weights and
+  // bias are having shape in NCHW order
+  if ((var_name == "X") &&
+      (expected_kernel_type.data_layout_ == framework::DataLayout::kMKLDNN) &&
+      (tensor.layout() != framework::DataLayout::kMKLDNN)) {
+    auto attrs = Attrs();
+    auto ar = paddle::framework::AttrReader(attrs);
+    const std::string data_layout = ar.Get<std::string>("data_layout");
```

Comment:
Why here use "data_layout" instead of "data_format", I saw in other PRs, here is data_format.

Reply:
@LeoZhao-Intel Good observation. I have not chosen those names *data_format* and *data_layout*. Yes all other ops got *data_format* , but batch norm is having *data_layout* as described here: https://www.paddlepaddle.org.cn/documentation/docs/en/1.4/api/layers.html#batch-norm

I agree that names are inconsistent, but perhaps there was a reason after this design choice. 
Maybe @dzhwinter or @zhangting2020 can comment further on this names inconsistency.